bundles/core/org.eclipse.smarthome.core.transform/src/main/java/org/eclipse/smarthome/core/transform/internal/profiles/TransformationProfile.java

```diff
@@ -0,0 +1,137 @@
+import org.eclipse.smarthome.core.transform.TransformationHelper;
+import org.eclipse.smarthome.core.transform.TransformationService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link StateProfile} which transforms a state value send from the handler to the framework based on the provided
+ * {@link ProfileTypeUID}. The direction from the framework toward the handler is left untouched.
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+@NonNullByDefault
+public class TransformationProfile implements StateProfile {
+
+    private static Logger logger = LoggerFactory.getLogger(TransformationProfile.class);
+
+    public static final String PROFILE_NAME = "Transformation";
+    private final ProfileTypeUID profileTypeUID;
+    private final TransformationService service;
+
+    private static String PATTERN_PARAM = "pattern";
+
+    @NonNullByDefault({})
+    private final String function;
+    @NonNullByDefault({})
+    private final String format;
+
+    private final ProfileCallback callback;
+
+    public TransformationProfile(ProfileCallback callback, ProfileContext context, TransformationService service,
+            ProfileTypeUID typeUID) {
+
+        this.service = service;
+        this.profileTypeUID = typeUID;
+        this.callback = callback;
+
+        Object paramValue = context.getConfiguration().get(PATTERN_PARAM);
+        logger.debug("Value for parameter '{}' is '{}'", PATTERN_PARAM, paramValue);
+        if (paramValue instanceof String) {
+            String pattern = (String) paramValue;
+            if (pattern.contains(TransformationHelper.FUNCTION_VALUE_DELIMITER)) {
+                function = pattern.split(TransformationHelper.FUNCTION_VALUE_DELIMITER)[0];
+                format = pattern.split(TransformationHelper.FUNCTION_VALUE_DELIMITER)[1];
+            } else {
+                logError();
+                function = null;
+                format = null;
+            }
+        } else {
+            logError();
+            function = null;
+            format = null;
+
+        }
+    }
+
+    private void logError() {
+        logger.error("Parameter '{}' has to be a String containing separator '{}'. Profile will be inactive.",
+                PATTERN_PARAM, TransformationHelper.FUNCTION_VALUE_DELIMITER);
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return profileTypeUID;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+        callback.handleCommand((Command) state);
+    }
+
+    @Override
+    public void onCommandFromItem(Command command) {
+        callback.handleCommand(command);
+    }
+
+    @Override
+    public void onCommandFromHandler(Command command) {
+        if (function == null || format == null) {
+            return;
```

Comment:
Does being "inactive" mean just not transforming anything, or rather making the system useless? 😉